core/src/main/java/hudson/security/Permission.java

```diff
@@ -274,22 +233,52 @@ public void setEnabled(boolean enable) {
     public boolean getEnabled() {
         return enabled;
     }
-    
+
+    @Override public int compareTo(@Nonnull Permission o) {
+        return getId().compareTo(o.getId());
+    }
+
+    /**
+     * Comparator that orders {@link Permission} objects based on their ID.
+     */
+    @Deprecated
+    public static final Comparator<Permission> ID_COMPARATOR = Comparator.comparing(Permission::getId);
+
+    /**
+     * Convert the ID representation into {@link Permission} object.
+     *
+     * @return
+     *      null if the conversion failed.
+     * @see #getId()
+     */
+    public static @CheckForNull Permission fromId(@Nonnull String id) {
+        return PermissionRegistry.getInstance().permissionFromId(id).orElse(null);
+    }
+
     /**
      * Returns all the {@link Permission}s available in the system.
      * @return
      *      always non-null. Read-only.
      */
     public static @Nonnull List<Permission> getAll() {
-        return ALL_VIEW;
+        return PermissionRegistry.getInstance().getPermissions();
     }
 
     /**
-     * All permissions in the system but in a single list.
+     * System property to enable or disable the {@code ADMINISTER} implies {@code RUN_SCRIPTS} feature flag.
+     * By default, this is enabled to match legacy behavior.
+     * @since TODO
      */
-    private static final List<Permission> ALL = new CopyOnWriteArrayList<Permission>();
+    public static final String ADMINISTER_IMPLIES_RUN_SCRIPTS = Permission.class.getName() + ".administerImpliesRunScripts";
 
-    private static final List<Permission> ALL_VIEW = Collections.unmodifiableList(ALL);
+    /**
+     * Indicates whether or not the {@code ADMINISTER} permission implies the {@code RUN_SCRIPTS} permission.
+     * @return true when ADMINISTER implies RUN_SCRIPTS, false when RUN_SCRIPTS implies ADMINISTER
+     * @since TODO
+     */
+    public static boolean administerImpliesRunScripts() {
```

Comment:
`@Restricted(NoExternalUse.class)` and delete `@since`, assuming there is no use case for calling this from plugins.

Reply:
Can do.